<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STM32 Image Flasher</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="RefreshAvailableBoards_pushButton">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="AvailableBoards_comboBox"/>
        </item>
        <item>
         <widget class="QPushButton" name="connect_pushButton">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="BrowseHexFile_pushButton">
          <property name="text">
           <string>Browse HEX File</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="communication_label">
            <property name="text">
             <string>Communication</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="communication_comboBox">
            <item>
             <property name="text">
              <string>UART</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>CAN</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="deviceId_label">
            <property name="text">
             <string>Device ID</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="deviceId_spinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="Flash_pushButton">
          <property name="text">
           <string>Flash Image</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="LoadedHexFile_label">
        <property name="text">
         <string>Loaded Hex File :</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="CurrentTask_label">
        <property name="text">
         <string>Current Task :</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="CurrentTask_progressBar">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableWidget" name="HexTable_tableWidget">
      <column>
       <property name="text">
        <string>Address</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Value</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPlainTextEdit" name="log_plainTextEdit">
      <property name="plainText">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
